cmd/wcc/main.go

```diff
@@ -0,0 +1,278 @@
+		}
+		fmt.Fprintf(os.Stderr, "Usage: %s <cmd> [<flags>]\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "\nAvailable commands are:\n\t%s\n", strings.Join(available, "\n\t"))
+		os.Exit(2)
+	}
+	run, ok := commands[os.Args[1]]
+	if !ok {
+		available := make([]string, 0, len(commands))
+		for name := range commands {
+			available = append(available, name)
+		}
+		fmt.Fprintf(os.Stderr, "Unknown command %q\n", os.Args[1])
+		fmt.Fprintf(os.Stderr, "\nAvailable commands are:\n\t%s\n", strings.Join(available, "\n\t"))
+		os.Exit(2)
+	}
+
+	// Skip two first as second argument is the command name that we just
+	// consumed.
+	if err := run(os.Stdin, os.Stdout, os.Args[2:]); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func cmdNewTransferProposal(input io.Reader, output io.Writer, args []string) error {
+	fl := flag.NewFlagSet("", flag.ExitOnError)
+	fl.Usage = func() {
+		fmt.Fprintln(flag.CommandLine.Output(), `
+Transfer funds from source account to the destination.
+		`)
+		fl.PrintDefaults()
+	}
+	var (
+		srcFl    = flAddress(fl, "src", "", "A source account address that the founds are send from.")
+		dstFl    = flAddress(fl, "dst", "", "A destination account address that the founds are send to.")
+		amountFl = flCoin(fl, "amount", "1 IOV", "An amount that is to be transferred between the source to the destination accounts.")
+		memoFl   = fl.String("memo", "", "A short message attached to the transfer operation.")
+		titleFl  = fl.String("title", "Transfer funds to distribution account", "The proposal title.")
+		descFl   = fl.String("description", "Transfer funds to distribution account", "The proposal description.")
+		eRuleFl  = fl.String("electionrule", "", "The ID of the election rule to be used.")
+	)
+	fl.Parse(args)
+
+	option := app.ProposalOptions{
+		Option: &app.ProposalOptions_SendMsg{
+			SendMsg: &cash.SendMsg{
+				Metadata: &weave.Metadata{Schema: 1},
+				Src:      *srcFl,
+				Dest:     *dstFl,
+				Amount:   amountFl,
+				Memo:     *memoFl,
+				Ref:      nil,
+			},
+		},
+	}
+	rawOption, err := option.Marshal()
+	if err != nil {
+		return fmt.Errorf("cannot serialize %T option: %s", option, err)
+	}
+	tx := &app.Tx{
+		Sum: &app.Tx_CreateProposalMsg{
+			CreateProposalMsg: &gov.CreateProposalMsg{
+				Metadata: &weave.Metadata{Schema: 1},
+				Base: &gov.CreateProposalMsgBase{
+					Title:          *titleFl,
+					Description:    *descFl,
+					StartTime:      weave.AsUnixTime(time.Now().Add(time.Minute)),
+					ElectionRuleID: []byte(*eRuleFl),
+				},
+				RawOption: rawOption,
+			},
+		},
+	}
+	raw, err := tx.Marshal()
+	if err != nil {
+		return fmt.Errorf("cannot serialize transaction: %s", err)
+	}
+	_, err = output.Write(raw)
+	return err
+}
+
+func flAddress(fl *flag.FlagSet, name, defaultVal, usage string) *weave.Address {
+	var a weave.Address
+	if defaultVal != "" {
+		var err error
+		a, err = weave.ParseAddress(defaultVal)
+		if err != nil {
+			panic(err.Error())
```

Comment:
pp: panics won't print a nice human readable error message. Why not print and os.Exit instead?

Reply:
I do not have a strong opinion about it. My thinking was that this (panic + err) will provide both an error message and the stack trace of where it happened. I am using `panic` only in truly unusual situations.

I can change it to `fmt.Fprintf(os.Stderr, "%+v", err)` so that if supported, the error stack trace will be printed instead. I think for the user it does not make the difference. It matters only for us programmers.
What do you think?

Reply:
Since this is just used in main.go, how about adding a `func die(err error)` which will end the program. And call this everywhere we abort.

Then we can discuss this in one place. Either

```go
func die(err error) {
  panic(err)
}
```

```go
func die(err error) {
  panic(err.Error())
}
```

```go
func die(err error) {
  fmt.Fprintf(os.Stderr, "%+v", err)
  os.Exit(1)
}
```

I kind of like the last one, but I think just having one function for termination will make this easy to change later if needed.

Reply:
I have implemented [`flagDie` function](https://github.com/iov-one/weave/blob/cli_create_transfer_issue_625/cmd/bnscli/flag.go#L85). It does `printf` and `exit 2`. Formatting of the message is left to the caller, but you cannot do much more than just print the error. Error formatting doesn't matter as they all come most of the time directly from the stdlib.